example/lib/screens/home.dart

```diff
@@ -1,10 +1,28 @@
 import 'package:flutter/material.dart';
 import 'package:airship_example/styles.dart';
-import 'package:airship_example/bloc/airship_bloc.dart';
 import 'package:airship_example/widgets/notifications_enabled_button.dart';
+import 'package:airship/airship.dart';
+
+class Home extends StatefulWidget {
+  @override
+  _HomeState createState() => _HomeState();
+}
 
 class _HomeState extends State<Home> {
-  final AirshipBloc _airshipBloc = AirshipBloc();
+  @override
+  void initState() {
+    initAirshipListeners();
+    super.initState();
+  }
+
+  // Platform messages are asynchronous, so we initialize in an async method.
+  Future<void> initAirshipListeners() async {
+    Airship.onChannelRegistration.listen((event) {
+      if (mounted) {
+        setState(() {});
```

Comment:
Can you also explain this one? Thanks for teaching me Dart.

Reply:
You can infer if the widget is still displaying by checking its mounted flag. In this case, we avoid trying to set the state of the widget when its unmounted (aka. no longer visible) since we can receive this async Airship callbacks at odd times in the widget's lifecycle.

Reply:
What does `() {}` do?

Reply:
setState notifies the UI that the internal state of the widget has changed. It has a block that can be supplied to key off specific properties on the widget, in this case we want to do a complete rebuild so the setter block is left blank.

Some dart people consider this a code smell. I think this is true for projects that need to key state changes carefully to prevent unnecessary refreshes; but I disagree that it's a problem in this instance because the potential performance hit is negligible to the point of being irrelevant in our use case.